src/Controller/MultiUploadController.php

```diff
@@ -2,66 +2,49 @@
-use Sonata\MediaBundle\Controller\MediaAdminController;
+use Sonata\MediaBundle\Admin\ORM\MediaAdmin;
 use Sonata\MediaBundle\Model\MediaInterface;
 use Sonata\MediaBundle\Provider\MediaProviderInterface;
+use Sonata\MediaBundle\Provider\Pool;
+use Symfony\Component\Form\FormFactoryInterface;
 use Symfony\Component\Form\FormInterface;
 use Symfony\Component\HttpFoundation\JsonResponse;
 use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
 
-/**
- * @codeCoverageIgnore
- */
-class MultiUploadController extends MediaAdminController
+final class MultiUploadController
 {
-    /**
-     * @var ManagerInterface
-     */
-    private $mediaManager;
-
-    public function __construct(ManagerInterface $mediaManager)
-    {
-        $this->mediaManager = $mediaManager;
-    }
-
-    public function createAction(Request $request = null)
-    {
-        $this->admin->checkAccess('create');
-
-        if (!$request->get('provider') && $request->isMethod('get')) {
-            $pool = $this->get('sonata.media.pool');
-
-            return $this->render('@SonataMultiUpload/select_provider.html.twig', [
-                'providers' => $pool->getProvidersByContext(
-                    $request->get('context', $pool->getDefaultContext())
-                ),
-                'action' => 'create',
-            ]);
-        }
-
-        return parent::createAction($request);
+    public function __construct(
+        private FormFactoryInterface $formFactory,
+        private ManagerInterface $mediaManager,
+        private MediaAdmin $mediaAdmin,
+        private Pool $mediaProviderPool,
+        private Responder $responder,
+        private int $maxUploadSize,
+        private string $redirectTo,
+    ) {
     }
 
-    public function multiUploadAction(Request $request)
+    public function __invoke(Request $request): Response
     {
-        $this->admin->checkAccess('create');
+        $this->mediaAdmin->checkAccess('create');
 
         $providerName = $request->query->get('provider');
         $context = $request->query->get('context', 'default');
 
-        /** @var MediaProviderInterface $provider */
-        $provider = $this->get($providerName);
+        $provider = $this->mediaProviderPool->getProvider($providerName);
 
         $form = $this->createMultiUploadForm($provider, $context);
         if (!$request->files->has('file')) {
-            return $this->render('@SonataMultiUpload/multi_upload.html.twig', [
+            return $this->responder->render('@SonataMultiUpload/multi_upload.html.twig', [
                 'action' => 'multi_upload',
                 'form' => $form->createView(),
                 'provider' => $provider,
-                'maxUploadFilesize' => $this->container->getParameter('sonata_multi_upload.max_upload_filesize'),
-                'redirectTo' => $this->container->getParameter('sonata_multi_upload.redirect_to'),
+                'maxUploadFilesize' => $this->maxUploadSize,
```

Comment:
```suggestion
                'maxUploadFilesize' => $this->maxUploadFilesize,
```
To match the key?
Just a proposal

Reply:
Well better to fix the key, the config is also called like that.